Seedvault: Redesign SUW activities for 12

BadDaemon opened the pull request:
* New SetupWizard uses external/setupdesign (sud) and
  external/setupcompat (suc)
  to achieve the desired looks
* It prevents proper previews when included in any app, though, so
  we better try achieving the same by manually styling the activities
  for 12
* This patch mimics the design by applying styles similar to the ones
  from sud, sometimes reusing parts taken from there

Change-Id: Iccf588e5239aeb00645f97b095910986f32c52d0

BadDaemon committed:
Seedvault: Redesign SUW activities for 12

* New SetupWizard uses external/setupdesign (sud) and
  external/setupcompat (suc) to achieve the desired looks
* It prevents proper previews when included in any app, though, so
  we better try achieving the same by manually styling the activities
  for 12
* This patch mimics the design by applying styles similar to the ones
  from sud, sometimes reusing parts taken from there

Change-Id: Iccf588e5239aeb00645f97b095910986f32c52d0

chirayudesai commented:
Screenshots:

<details>
<summary>Light</summary>

![image](https://user-images.githubusercontent.com/952062/166522489-94b53547-203e-4343-a73e-a32460c946c4.png)
![image](https://user-images.githubusercontent.com/952062/166522501-01b77ade-0f61-4dc9-8ce4-99acd0238f17.png)
![image](https://user-images.githubusercontent.com/952062/166522509-e39f078d-7a2c-44e3-ba0a-58a3b8436631.png)
![image](https://user-images.githubusercontent.com/952062/166522519-e4655f3f-534c-426b-bbbe-60a30595e8de.png)
![image](https://user-images.githubusercontent.com/952062/166522532-6f77590f-140e-4a8d-95e2-ebb59792ce6c.png)
</details>

<details>
<summary>Dark / Black:</summary>

![image](https://user-images.githubusercontent.com/952062/166522582-3f869361-fd7b-4541-8b93-2140c721ddf5.png)
![image](https://user-images.githubusercontent.com/952062/166522599-213cc7ee-3b37-4e41-aceb-20a041a7f7f1.png)
![image](https://user-images.githubusercontent.com/952062/166522604-d078c1fc-634d-4c7f-8274-9fbfb55552e5.png)
![image](https://user-images.githubusercontent.com/952062/166522615-08cd0966-d540-4c7a-aea3-d712bcd21d68.png)
![image](https://user-images.githubusercontent.com/952062/166522632-e4edf83e-e6d9-4ee5-89a1-591ba9fb22d0.png)
![image](https://user-images.githubusercontent.com/952062/166522644-62d991b5-7806-4003-8c4f-f632d05a5960.png)
![image](https://user-images.githubusercontent.com/952062/166522654-c295e3dc-da68-4980-a852-56186e5943d0.png)

</details>

grote commented:
Thanks a lot @BadDaemon that's great stuff! I notice that the file/backup lists now look a bit squeezed with big margin. Is this intentional?

Can you give me a week to look into the diff details?

BadDaemon commented:
I fixed a few issues:
a) Top margin was wrong, it now matches the actual SetupDesign
b) The items in the lists were indeed squeezed too much since I applied the default margin AND the items had padding.
    I removed the padding for these cases
c) I couldn't get rid of the scrollbars. Moving them in with the margin made it look weird. Therefore I applied the right margin as a padding to the items instead so the scrollbars stay on the right
d) Fixed statusbar background color to match the window's

Screenshots:
<details>
<summary>Light</summary>

![Light1](https://user-images.githubusercontent.com/17656887/167253494-1577446c-0267-433b-961d-8705250e1562.png)
![Light2](https://user-images.githubusercontent.com/17656887/167253500-e8aa234a-87e2-4595-a91f-869685cd189f.png)
![Light3](https://user-images.githubusercontent.com/17656887/167253502-9c25ebe0-a088-4141-83d1-6e4080fda461.png)
![Light4](https://user-images.githubusercontent.com/17656887/167253503-fe4f0f5e-64fd-4566-85f9-a099a9f15287.png)

</details>

<details>
<summary>Dark/Black</summary>

![Dark1](https://user-images.githubusercontent.com/17656887/167253521-2c5f37e3-90b0-4643-a711-7a62c5044724.png)
![Dark2](https://user-images.githubusercontent.com/17656887/167253525-186104ce-0356-4e1c-97de-772b429d0b0f.png)
![Dark3](https://user-images.githubusercontent.com/17656887/167253526-99522aec-4852-4c01-aff5-6478180b009b.png)
![Dark4](https://user-images.githubusercontent.com/17656887/167253527-96c0d570-5306-4ebe-9e82-ca57529dc388.png)

</details>

BadDaemon commented:
e) Now also fixed the navigation bar color

grote commented on app/src/main/java/com/stevesoltys/seedvault/restore/RestoreProgressFragment.kt:
```diff
@@ -54,7 +54,7 @@ class RestoreProgressFragment : Fragment() {
             layoutManager = this@RestoreProgressFragment.layoutManager
             adapter = this@RestoreProgressFragment.adapter
-            addItemDecoration(DividerItemDecoration(context, VERTICAL))
+            addItemDecoration(DividerItemDecoration(context, SHOW_DIVIDER_NONE))
```
Could we also just not add any item decoration or would this look different?

BadDaemon replied:
will see

grote commented on app/src/main/res/layout/footer_snapshots.xml:
```diff
@@ -1,17 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
```
Nitpick: I don't know how you edit those files, but it would be great if you could use the Android Studio XML auto-format, so in later MRs, there's no unrelated changed introduced in the diffs. I comment here, where it is obvious, but also the sort order below is changed by Android Studio.

BadDaemon replied:
will look into this. i did it mostly because I don't like the different styles where some elements have their start tag standalone and thngs like this have the scheme in the same line (but a second scheme standalone again). aesthetics. not essential or relevant to the change, ofc

grote commented:
What about the fragment where the user needs to enter their recovery code? Was this left intentionally in the old design?

grote commented:
What about the files backup items where the user needs to select which backup to restore? Was this left intentionally in the old design?

grote commented on app/src/main/res/layout/fragment_restore_progress.xml:
```diff
@@ -68,17 +68,14 @@
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintHorizontal_bias="1.0"
-        app:layout_constraintStart_toStartOf="parent" />
+        app:layout_constraintTop_toBottomOf="@id/appList" />
```
`+` missing?

BadDaemon replied:
the '+' in these places isn't correct. it's supposed to be used to create the id, which is done on the id tag of the actual element. references should use it without the '+'. not essential to the change, though

grote commented on app/src/main/res/layout/fragment_restore_set.xml:
```diff
@@ -19,22 +18,20 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/imageView" />
+        app:layout_constraintTop_toBottomOf="@id/imageView" />
```
Why was the `+` removed?

BadDaemon replied:
see above

grote commented on app/src/main/res/layout/fragment_restore_set.xml:
```diff
@@ -19,22 +18,20 @@
-        android:layout_marginStart="16dp"
-        android:layout_marginTop="16dp"
-        android:layout_marginEnd="16dp"
+        android:layout_marginEnd="0dp"
```
It feels a bit strange use the new `SudContent` style which adds margin start and end, but then set only the end to 0 here and in some other places. why is this?

BadDaemon replied:
this is due to

c) I couldn't get rid of the scrollbars. Moving them in with the margin made it look weird. Therefore I applied the right margin as a padding to the items instead so the scrollbars stay on the right

grote commented on app/src/main/res/layout/list_item_storage_root.xml:
```diff
@@ -41,7 +41,7 @@
         android:layout_height="wrap_content"
         android:layout_marginTop="8dp"
-        android:layout_marginEnd="16dp"
+        android:layout_marginEnd="40dp"
```
I think this is the answer why you set the end margin of the recyclerview to 0, because the padding it is the items. To be honest, I think it is strange that this is mixed like this. The end padding comes from the items, but the start from the list? Is this an SUW thing?

BadDaemon replied:
same as above, c)

grote commented on app/src/main/res/layout/list_item_restore_set.xml:
```diff
@@ -11,7 +12,6 @@
         android:id="@+id/imageView"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginStart="16dp"
```
removing this margin here causes the ripple affect to have a sharp border with the icon.

BadDaemon replied:
will look into this

grote reviewed (changes requested):
Looks good overall. Just some few nitpicks above. It would be great if the padding/margin on lists could be applied a bit more consistently.

BadDaemon commented:
> What about the files backup items where the user needs to select which backup to restore? Was this left intentionally in the old design?

i only changed those i was able to actually see and verify. need to check how i can get those two to show and will adjust them accordingly

grote commented on storage/lib/src/main/res/layout/item_snapshot.xml:
```diff
@@ -41,11 +42,10 @@
+            app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toEndOf="@+id/timeView"
-            app:layout_constraintTop_toBottomOf="@+id/nameView"
+            app:layout_constraintTop_toTopOf="parent"
```
I wonder if with those changes, a long name could go over the size?

Edit: Ah no, the above takes care of this.

grote commented:
Thanks for the patience! Changes look good.
